CODING.md

```diff
@@ -222,6 +222,15 @@ The CI tooling will check and enforce the order and required sections when a spr
   * When no golang code is involved, the context prefix before the colon can refer to directories or top-level files instead.
     * `build-aux,.github/workflows: limit make processes with nproc`
 
+* Merging
```

Comment:
Modified CONTRIBUTING as well, also removed `need-architecture-review` label that we decided to remove because it was similar to `Needs Samuele review`.